### ISSUE TITLE
Question about memory

### ISSUE THREAD

**Issue by jayflo** [jeffbski/redis-wstream#8]
The Readme says:
> Client should not be a normal (non-multi batch) client otherwise it will accumulate the stream in memory before sending.

But when I look at the npm package https://www.npmjs.com/package/redis documentation under `batch` it says

> Please remember that all commands are kept in memory until they are fired.

Do you mean that you *should* use a "normal" client?  This would also agree with the example in the README (which is a "normal" client).

**Comment by binarymist** [comment]
The way I read this is that Client should not be a [multi](https://github.com/NodeRedis/node_redis#clientmulticommands) or [batch](https://github.com/NodeRedis/node_redis#clientbatchcommands) client, as both will be
> queued up until `EXEC` is issued

Which as I understand it, completly defeats the point of streaming?
> The only difference between .batch and .multi is that no transaction is going to be used

on `batch` that is.

I think what the README means is: Client should not be a multi or batch client otherwise it will accumulate the stream in memory before sending.

If this wan't the case, it wouldn't make any sense.

As it stands, the [Why](https://github.com/jeffbski/redis-wstream#why) section of the README says...
> I would prefer to use an existing connection

Which in the default case won't be a multi or batch, which would defeat the purpose of the Why statement

If this is not correct, please provide some clarity, or even better, plese update the README so that it is more obvious?

Thanks.

**Comment by jeffbski** [comment]
Yes I believe @binarymist has explained it properly. Last time I investigated the redis batch clients they seemed to pool up the data locally before sending so you could end up with a really large blob of memory to send all at once one the stream ended.

However if you use a non-multi non-batch client then the chunks are sent as they are being received, so you are sending the data to redis quickly as each chunk is read. Thus you are streaming it to redis over time rather than sending the whole thing at once. If these files are large then you wouldn't need as much RAM to send it since it would be sending in smaller chunks.
